### PR TITLE
(#4106) Fix role permission assigning

### LINKED ISSUE #4106: The newest version of role_delegation deletes roles from site admin
This occurs only under some unknown conditions. See https://www.drupal.org/project/role_delegation/issues/3263782#comment-15271722 for more information, but the TL;DR; is that we really don't need this. The Site Admin role should be able to assign all roles in the system. There are no roles it cannot assign, so this module is not exactly needed. What we can do is make a new permission and alter the user form to change the field access for the role field to allow a user with that permission to change roles. The role_delegation module does this in a fashion, but makes it so that each grantable role is a separate permission. 

FYI, OOB Drupal only checks for the "administer permissions" permission for setting roles on a user. This however, also allows those users to grant any permission to any role, and even create new roles. This would obviously be bad in a production system.

### PR DESCRIPTION
**Pull request by bryanpizzillo**
role_delegation removes all roles from site admin under certain conditions. This disables the module and provides an alternate method.

Closes #4106

### CODE REVIEW

**Comment by KateMashkinaNIH** [comment]
no option to filter users on the people tab
Screenshot 2023-10-16 at 10 14 10 AM

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module [hunk 1]
```diff
@@ -207,6 +207,14 @@ function cgov_core_form_alter(&$form, FormStateInterface $form_state, $form_id)
...
     }
   }
+
+  if ($form_id === 'user_form') {
+    if (isset($form['account']['roles'])) {
+      $user = \Drupal::currentUser();
+      $form['account']['roles']['#access'] = $form['account']['roles']['#options'] &&
+        ($user->hasPermission('administer permissions') || $user->hasPermission('edit user roles'));
+    }
+  }
```
Comment (blairlearn) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module:217]: This needs a comment to explain what it's doing and why.

Also, the hook's comment block is no longer correct.

**Review by blairlearn — CHANGES_REQUESTED** [review]

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/CgovCoreTools.php [hunk 2]
```diff
@@ -238,6 +238,30 @@ public function addRolePermissions(array $rolePermissions) {
...
+    $role_storage = $this->entityTypeManager->getStorage('user_role');
+    $roles = $role_storage->loadMultiple(array_keys($rolePermissions));
+
+    // Add all permissions.
+    foreach ($rolePermissions as $roleId => $permissionId) {
+      foreach ($permissionId as $perm) {
+        $roles[$roleId]->revokePermission($perm);
+      }
+      $roles[$roleId]->save();
+    }
```
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/CgovCoreTools.php]: Removed comment

**Comment by bryanpizzillo** [comment]
> no option to filter users on the people tab Screenshot 2023-10-16 at 10 14 10 AM

Yeah this was because `role_delegation` did some altering of the people view, which made it a dependency. When the module was uninstalled it deleted the view. Since this view is part of core it is not part of our configuration and was not something that would be re-installed by Features.

This has been addressed in the latest push.

**Comment by KateMashkinaNIH** [comment]
this passes QA

**Review by blairlearn — APPROVED** [review]

**Pull request merged**